From 7245ad7bd9e5caafafe9162f54fa527859eaf6ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Sun, 14 Jul 2013 00:44:05 -0400
Subject: [PATCH 3/3] gnome-ask-password-agent: report GLib.Error only to
 stderr

When a GLib.Error happens, it is likely that showing a message box
would fail too.

Solution copied from systemadm and
https://bugzilla.redhat.com/show_bug.cgi?id=716663.

http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=716369
---
 src/gnome-ask-password-agent.vala | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/gnome-ask-password-agent.vala b/src/gnome-ask-password-agent.vala
index 5df27ee..0f6284d 100644
--- a/src/gnome-ask-password-agent.vala
+++ b/src/gnome-ask-password-agent.vala
@@ -256,9 +256,10 @@ int main(string[] args) {
 
                 MyStatusIcon i = new MyStatusIcon();
                 Gtk.main();
-
-        } catch (GLib.Error e) {
+        } catch (IOError e) {
                 show_error(e.message);
+        } catch (GLib.Error e) {
+                Posix.stderr.printf("%s\n", e.message);
         }
 
         return 0;
-- 
1.8.3.4

